Assets/BossRoom/Scripts/Client/Game/State/ClientMainMenuState.cs

```diff
@@ -100,7 +100,7 @@ void OnAuthSignIn()
 
             void OnSignInFailed()
             {
-                Debug.LogError("For some reason we can't authenticate the user anonymously - that typically means that project is not properly set up with Unity services.");
+                PopupPanel.ShowPopupPanel("Authentication Error", "For some reason we can't authenticate the user anonymously - that typically means that project is not properly set up with Unity services.");
```

Comment:
I'm thinking this, and `LobbyUIMediator.OnRelayJoinFailed`'s invocation of PopupPanel could serve instead as places where a "Relay" message is published, instead of Popup panel being invoked in these two places. Granted, there's only 2 instances of Relay-based connection messages, but it would be a neat place to wrap up these two message strings that shouldn't live inside `ClientMainMenuState` or `LobbyUIMediator`.

Reply:
There currently exists a UnityServiceErrorMessage that is used to publish messages linked with Lobby. I could use this same type to publish messages for Authentication and Relay as well, and add a subscriber to this channel to call the PopupPanel, which would also add popups for those other Lobby messages. However, some of the Lobby messages happen at the same time as connection status messages (i.e.: receive a GenericDisconnect from the host shutting down, then a "Host has left the lobby" message), so to avoid overwriting the original popup we would need to add popup stacking. Unless we think it should be in a later PR, I'll refactor this to allow stacking

Reply:
I see. So if stacking not going in for this PR, this is okay. Is there a ticket already for stacking popup messages?

Reply:
yes, and another one to redirect those messages to the popup panel

Reply:
Beautiful 👍